File: STL/Container/stack.cpp
#include<bits/stdc++.h>
using namespace std;

void print(stack<int> s){
	stack<int>temp=s;
	while(!s.empty()){
		cout<<s.top()<< "    ";
		s.pop();
	}
	cout<<endl;
}

int main(){

	stack<int>s;
	s.top();
	/**
	 * empty() returns true if the stack is empty
	 * size() returns the size of the stack
	 * top() returns the topmost element of the stack
	 * push() pop() the usual insertion and deletion 
	 * swap() swaps the the stack of the same type and size can vary
	 * emplace() is another name for inserting in the top
	 */


	return 0;
}

File: JavaCollections/stack.java
import java.util.*;
public class stack{
	public static void main(String[] args) {
		Stack<String> words= new Stack<>();

		//	push(Object o) : it pushes the Object into the stack
		//	pop() : it is used to remove the topmost element in the stack
		words.push("this");
		words.push("this");
		words.push("this");

		//	peek()	: returns the topmost element in the stack 
		System.out.println("the top most element in the list : "+words.peek());
		words.pop();
		System.out.println("the top most element in the list : "+words.peek());
		System.out.println(words.size());

		System.out.println(words);
	}
}